numpydoc/numpydoc.py

```diff
@@ -184,6 +183,33 @@ class NumpyCDomain(ManglingDomainBase, CDomain):
     }
 
 
+def match_items(lines, content_old):
+    """Create the right items for lines.
```

Comment:
I found myself digging into the docutils source to figure out what these "right 'items'" are.  Can we clear up the docstring of this function a bit, describing what it is doing and why?  I see some of that below, but it's not easy to understand without the context of the PR.